internal/s3shared/arn/arn.go

```diff
@@ -48,6 +48,10 @@ func ParseResource(s string, resParser ResourceParser) (resARN Resource, err err
 		return nil, InvalidARNError{ARN: a, Reason: "service is not supported"}
 	}
 
+	if strings.HasPrefix(a.Region, "fips-") || strings.HasSuffix(a.Region, "-fips") {
+		return nil, InvalidARNError{ARN: a, Reason: "FIPS region not allowed in ARN"}
+	}
+
```

Comment:
1. Had to do copy this to avoid the circular dependency between `s3shared` and `arn` package. Creating a separate package for this pretty simple check seemed overkill.
2. Yeah we can move this check higher up.

Reply:
Hmm, on second thought regarding pointer number 2, I think keeping this at the s3 level is right for now. It's to avoid a very limited situation where a user  hand-construct an ARN themselves, and we want to ensure the correct intent in this situation where the ARN region has specific meaning and behavior.

Reply:
Keeping at S3 level is fine for now. Though I don't follow how user hand-constructing an ARN would affect anything in this case.